python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -751,6 +751,119 @@ interface {{ ethernet_interface.name }}
+{%             endif %}
+   {{ isis_auth_cli }} level-1
+{%         endif %}
+{%         if ethernet_interface.isis_authentication.level_2.mode is arista.avd.defined
+               and (ethernet_interface.isis_authentication.level_2.mode in ["md5", "text"]
+               or (ethernet_interface.isis_authentication.level_2.mode == "sha" and ethernet_interface.isis_authentication.level_2.sha.key_id is arista.avd.defined)
+               or (ethernet_interface.isis_authentication.level_2.mode == "shared-secret"
+                   and ethernet_interface.isis_authentication.level_2.shared_secret.profile is arista.avd.defined
+                   and ethernet_interface.isis_authentication.level_2.shared_secret.algorithm is arista.avd.defined)) %}
+{%             set isis_auth_cli = "isis authentication mode " ~ ethernet_interface.isis_authentication.level_2.mode %}
+{%             if ethernet_interface.isis_authentication.level_2.mode == "sha" %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " key-id " ~ ethernet_interface.isis_authentication.level_2.sha.key_id %}
+{%             elif ethernet_interface.isis_authentication.level_2.mode == "shared-secret" %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " profile " ~ ethernet_interface.isis_authentication.level_2.shared_secret.profile ~
+                      " algorithm " ~ ethernet_interface.isis_authentication.level_2.shared_secret.algorithm %}
+{%             endif %}
+{%             if ethernet_interface.isis_authentication.level_2.rx_disabled is arista.avd.defined(true) %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " rx-disabled" %}
+{%             endif %}
+   {{ isis_auth_cli }} level-2
+{%         endif %}
+{%     endif %}
+{%     if ethernet_interface.isis_authentication is arista.avd.defined %}
+{%         set both_key_ids = [] %}
+{%         if ethernet_interface.isis_authentication.both.key_ids is arista.avd.defined %}
+{%             for auth_key in ethernet_interface.isis_authentication.both.key_ids | arista.avd.natural_sort("id") %}
+{%                 if auth_key.id is arista.avd.defined
+                       and auth_key.algorithm is arista.avd.defined
+                       and auth_key.key_type is arista.avd.defined
+                       and auth_key.key is arista.avd.defined %}
+{%                     do both_key_ids.append(auth_key.id) %}
+{%                     if auth_key.rfc_5310 is arista.avd.defined(true) %}
+   isis authentication key-id {{ auth_key.id }} algorithm {{ auth_key.algorithm }} rfc-5310 key {{ auth_key.key_type }} {{ auth_key.key }}
+{%                     else %}
+   isis authentication key-id {{ auth_key.id }} algorithm {{ auth_key.algorithm }} key {{ auth_key.key_type }} {{ auth_key.key }}
+{%                     endif %}
+{%                 endif %}
+{%             endfor %}
+{%         endif %}
+{%         if ethernet_interface.isis_authentication.level_1.key_ids is arista.avd.defined %}
```

Comment:
i see the diff in this logic compare to isis vlan, can you please check. Also if there is required or primary key we may not need to add additional check now

Reply:
Only natural_sort was missing. I added it. Now the logic is same as VLAN int.